### ISSUE TITLE
Documentation - 1.15.x compatibility notes - Default Envoy Cipher Suite changes

### ISSUE THREAD

**Issue by tcoffin** [istio/istio#42648]
### Bug Description

With Istio 1.15.x the default cipher suite for Envoy has changed.

Istio **1.15.x** uses Envoy **1.23.x** 
Istio **1.14.x** uses Envoy **1.22.x**

[Envoy 1.23.x default cipher suites (boringssl, non-fips)](https://www.envoyproxy.io/docs/envoy/v1.23.3/api-v3/extensions/transport_sockets/tls/v3/common.proto.html):
```
[ECDHE-ECDSA-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]
[ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POLY1305]
ECDHE-ECDSA-AES256-GCM-SHA384
ECDHE-RSA-AES256-GCM-SHA384
```

[Envoy 1.22.x default cipher suites (boringssl, non-fips)](https://www.envoyproxy.io/docs/envoy/v1.22.5/api-v3/extensions/transport_sockets/tls/v3/common.proto.html):
```
[ECDHE-ECDSA-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]
[ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POLY1305]
ECDHE-ECDSA-AES128-SHA
ECDHE-RSA-AES128-SHA
AES128-GCM-SHA256
AES128-SHA
ECDHE-ECDSA-AES256-GCM-SHA384
ECDHE-RSA-AES256-GCM-SHA384
ECDHE-ECDSA-AES256-SHA
ECDHE-RSA-AES256-SHA
AES256-GCM-SHA384
AES256-SHA
```

In summary, Envoy 1.23 looks to drop support for ciphers that include: SHA-1, non-ephemeral keys, and CBC
These are overall stronger cipher choices, however it has potential impacts to legacy client connections.

### Version

```prose
$ istioctl version
client version: 1.15.4
control plane version: 1.15.4
data plane version: 1.15.4 (97 proxies)

$ kubectl version --short
Client Version: v1.25.2
Kustomize Version: v4.5.7
Server Version: v1.22.15-eks-fb459a0
```

**Comment by kalioz** [comment]
Hello ! 
This change impacted us (Gateway allowing TLS1_0 and TLS1_1, no additional ciphers added), and we had a hard time figuring out (1) what happened and (2) how to fix it to make it work again.
A warning has been added to the Gateway deployment to warn if there's no ciphers, but the doc doesn't include which ciphers were allowed, not which one to use for this kind of (deprecated) TLS; basically, the `minProtocolVersion` field became silently inactive due to the Istio upgrade.

Could this be added to the release notes of Istio 1.15 ?

FWIW, to make it work again I've currently added all allowed ciphers on the Gateway, like such - a lot of these ciphers should be removed, but we need to find out which one our legacy clients use before removing the others :)  :
```yaml
apiVersion: networking.istio.io/v1beta1
kind: Gateway
spec:
  servers:
  - hosts:
    - myhost.com
    tls:
      cipherSuites:
      - ECDHE-ECDSA-AES256-GCM-SHA384
      - ECDHE-RSA-AES256-GCM-SHA384
      - ECDHE-ECDSA-AES128-GCM-SHA256
      - ECDHE-RSA-AES128-GCM-SHA256
      - AES256-GCM-SHA384
      - AES128-GCM-SHA256
      - ECDHE-ECDSA-CHACHA20-POLY1305
      - ECDHE-RSA-CHACHA20-POLY1305
      - ECDHE-ECDSA-AES128-SHA
      - ECDHE-RSA-AES128-SHA
      - ECDHE-ECDSA-AES256-SHA
      - ECDHE-RSA-AES256-SHA
      - AES128-SHA
      - AES256-SHA
      - DES-CBC3-SHA
      maxProtocolVersion: TLSV1_3
      minProtocolVersion: TLSV1_0
```
